lib/hooks/userspace/hook-mysql.js

```diff
@@ -0,0 +1,68 @@
+/**
+ * Copyright 2015 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+var cls = require('../../../lib/cls.js');
+var shimmer = require('shimmer');
+var semver = require('semver');
+var agent;
+
+var SUPPORTED_VERSIONS = '2.9.x';
```

Comment:
I believe it can be relaxed as far as `>= 2.x`. It is more a philosophical question of how/when we want to catch hook breaking errors. The current form makes it more likely that they're caught in our unit tests and less likely they would be reported as user facing errors. I'm happy to change it if we're ok with less test coverage of supported versions.